contracts/currency/currency.hpp

```diff
@@ -87,7 +87,8 @@ namespace TOKEN_NAME {
    /**
    Assert statement to verify structure packing for account
    **/
-   static_assert( sizeof(account) == sizeof(uint64_t)+sizeof(currency_tokens), "unexpected packing" );
+   //This structure is *not* packed the check will not work for different sizes of currency_tockens 
+   //static_assert( sizeof(account) == sizeof(uint64_t)+sizeof(currency_tokens), "unexpected packing" );
```

Comment:
the `master` branch doesn't currently deal with ambiguously packed structure very well.  Does packing the account/transfer etc structs not resolve this without relaxing this static_assert?

Reply:
Hi @wanderingbort , Yes first though was to pack it as the assert implies it checks for packed structure. However , packed structures are a bit unsafe when using them ( outside of WAST domain ) 
https://stackoverflow.com/questions/8568432/is-gccs-attribute-packed-pragma-pack-unsafe
I just did not want to introduce a bug , however I am ready to look into that and see if in existing code at least it will produce same webassembly.

Reply:
This is a wart for the current release that we have largely addressed in the future releases.   Don't feel too bad about it for now, we will handle merging the intent of your changes into the new version of our standard currency contract.

Reply:
As I suspected when I added struct PACKED(account) LLVM produced extra align instruction align=1 for WAST

before 
```
(i64.load offset=8
```
after packing 

``` 
(i64.load offset=8 align=1
```
So if you don't mind I will not PACK account or transfer for now and just change my comment to 
//TODO: reintroduce static_assert if account and transfer is packed in the future 

This way it's safe with my pull request  as I see quite few use cases when from.ballance and to.ballance are used , currently token<uint64_t , N(currency)> will be aligned correctly but if other types are mixed together then it could lead to a bug with PACKED(account)
Thank you for letting me work on EOS stack.